aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
-.. include:: /../common/stub-topic.txt
+Filters
+=======
 
-|stub-icon| Filters
-===================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+*Filters* in ASP.NET MVC allow you to run code before or after a particular stage in the execution pipeline. Filters can be configured globally, per-controller, or per-action.
 
-.. _issue: https://github.com/aspnet/Docs/issues/119
+.. contents:: Sections
+    :local:
+    :depth: 1
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/master/mvc/controllers/filters/sample>`_.
+
+How do filters work?
+--------------------
+
+Each kind of filter will be executed at a different stage in the pipeline, and thus has its own set of intended scenarios. Choose what kind of filter to create based on the task you need it to perform, and where in the request pipeline it executes. Filters run within the MVC Action Invocation Pipeline, sometimes referred to as the *Filter Pipeline*, which runs after MVC selects the action to execute.
+
+.. image:: filters/_static/filter-pipeline-1.png
+
+Different kinds of filters run at different points within the pipeline. Some filters, like authorization filters, only run before the next stage in the pipeline, and take no action afterward. Other filters, like action filters, can execute both before and after other parts of the pipeline execute, as shown below.
```

Comment:
Filter types run at specific points in the pipeline. Authorization filters run first.
[What does "and take no action afterward." - they run first and don't run in later stages?- if so, you don't need to say that]
Action filters can execute both before and after other parts of the pipeline execute, as shown below.